src/github.com/getlantern/peerscanner/host.go

```diff
@@ -80,33 +87,75 @@ func (h *host) String() string {
  * API for interacting with host
  ******************************************************************************/
 
-// newHost creates a new host for the given name, ip and optional DNS record.
-func newHost(name string, ip string, record *cloudflare.Record) *host {
-	// Cache TLS sessions
-	clientSessionCache := tls.NewLRUClientSessionCache(1000)
-
+// newHost creates a new host for the given name, ip and optional DNS records.
+func newHost(name string, ip string, port string, cflRecord *cloudflare.Record, dspRecord *dnsimple.Record) *host {
 	h := &host{
 		name:         name,
 		ip:           ip,
-		record:       record,
+		port:         port,
+		cflRecord:    cflRecord,
+		dspRecord:    dspRecord,
 		resetCh:      make(chan string, 1000),
 		unregisterCh: make(chan interface{}, 1),
 		statusCh:     make(chan chan *status, 1000),
+		initCfrCh:    make(chan interface{}, 1),
+	}
+
+	if h.isFallback() {
+
+		h.cflGroups = map[string]*cflGroup{
+			RoundRobin: &cflGroup{subdomain: RoundRobin},
+			Fallbacks:  &cflGroup{subdomain: Fallbacks},
+			Peers:      &cflGroup{subdomain: Peers},
+		}
+		h.dspGroups = map[string]*dspGroup{
+			RoundRobin: &dspGroup{subdomain: RoundRobin},
+			Fallbacks:  &dspGroup{subdomain: Fallbacks},
+			Peers:      &dspGroup{subdomain: Peers},
+		}
+		country := fallbackCountry(name)
+		if country != "" {
+			// Add host to country-specific rotation
+			h.cflGroups[country] = &cflGroup{subdomain: country}
+			h.dspGroups[country] = &dspGroup{subdomain: country}
+		}
+	} else {
+		log.Errorf("Somehow adding peer host? %v (%v)", name, ip)
+	}
+
+	return h
+}
+
+func (h *host) resetProxiedClient(port string) {
+
+	var dial func(addr string) (net.Conn, error)
+	if port == "80" {
+		dialer := net.Dialer{Timeout: dialTimeout}
```

Comment:
I assumed that the point of tls.NewLRUClientSessionCache(1000) was that it was worth reusing across calls to `dial` and that's why it was created at a scope such that it would remain the same across such calls.  Isn't that the thing whereby you avoid having to redo the whole TLS handshake on every connection?

Also, I made this thing uglier trying to accomodate the ports that servers report.  We can probably mandate port 80 (or 443 if testing with domain fronting), and dispense with that.  Then the proxiedClient can be created on host creation and stay unchanged for its lifetime, which is less confusing and keeps the cache working.

Re: pause, in principle I agree, but I haven't given it much thought in the context of this PR.  I don't suppose that's urgent to address, so let's hear Ox's rationale when he's back.